deep-learning/src/test/python/synapsemltest/dl/test_deep_vision_classifier.py

```diff
@@ -0,0 +1,113 @@
+# Copyright (C) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See LICENSE in project root for information.
+
+import contextlib
+import os
+import shutil
+import tempfile
+
+import pytest
+from horovod.spark.common.store import LocalStore
+from pyspark.ml.evaluation import MulticlassClassificationEvaluator
+from pyspark.sql import SparkSession
+from pyspark.sql.functions import col
+from pyspark.sql.types import DoubleType
+from pytorch_lightning.callbacks import ModelCheckpoint
+from synapse.ml.dl import *
+
+from .test_deep_vision_model import MyDummyCallback
+
+
+@contextlib.contextmanager
+def tempdir():
+    dirpath = tempfile.mkdtemp()
+    try:
+        yield dirpath
+    finally:
+        shutil.rmtree(dirpath)
+
+
+@contextlib.contextmanager
+def local_store():
+    with tempdir() as tmp:
+        store = LocalStore(tmp)
+        yield store
+
+
+def generate_data(spark, train_folder, test_folder):
+    train_files = [
+        os.path.join(dp, f)
+        for dp, dn, filenames in os.walk(train_folder)
+        for f in filenames
+        if os.path.splitext(f)[1] == ".jpg"
+    ]
+    test_files = [
+        os.path.join(dp, f)
+        for dp, dn, filenames in os.walk(test_folder)
+        for f in filenames
+        if os.path.splitext(f)[1] == ".jpg"
+    ]
+
+    def extract_path_and_label(path):
+        num = int(path.split("/")[-1].split(".")[0].split("_")[1])
+        label = num // 81  # Assign the label
+        return (path, label)
+
+    train_df = spark.createDataFrame(
+        map(extract_path_and_label, train_files), ["image", "label"]
+    ).withColumn("label", col("label").cast(DoubleType()))
+
+    test_df = spark.createDataFrame(
+        map(extract_path_and_label, test_files), ["image", "label"]
+    ).withColumn("label", col("label").cast(DoubleType()))
+
+    return train_df, test_df
+
+
+class CallbackBackend(object):
+    def run(self, fn, args=(), kwargs={}, env={}):
+        return [fn(*args, **kwargs)] * self.num_processes()
+
+    def num_processes(self):
+        return 1
+
+
+@pytest.mark.skip(reason="not testing this for now")
+def test_mobilenet_v2(get_data_path):
+    spark = SparkSession.builder.master("local[*]").getOrCreate()
+
+    ctx = CallbackBackend()
+
+    epochs = 5
+    callbacks = [
+        MyDummyCallback(epochs),
+        ModelCheckpoint(dirpath="target/mobilenet_v2/"),
+    ]
+
+    train_folder, test_folder = get_data_path
+
+    with local_store() as store:
+
+        deep_vision_classifier = DeepVisionClassifier(
+            backbone="mobilenet_v2",
+            store=store,
+            backend=ctx,
```

Comment:
Is this test skipped for a reason?

Reply:
emmm yes, perhaps I could also delete this test. I wrote this test for local tests, but I don't have gpus locally so can't enabling the gpu test, same thing for pipeline I guess. Another reason is that this test looks exactly the same as the notebook. So I skip this test in case others want to try with local machines containing gpus.

Reply:
ok so this is kept due to the simplicity of workign with it, makes sense